### DISCUSSION TITLE
[Bug]: Unknown field "spec.kafka.metadataVersion" when applying Kafka CR with KRaft and dual-role nodes in v1beta2 API version

### DISCUSSION THREAD

**Discussion by adababys** [strimzi/strimzi-kafka-operator#10975] [General]
### Bug Description

I encountered an issue while trying to apply a Kafka Cluster Resource (CR) with KRaft mode and dual-role nodes using the v1beta2 API version. The application of the YAML configuration fails due to an unknown field spec.kafka.metadataVersion.

### Steps to reproduce

1. Deployed Strimzi Kafka Operator ( strimzi-kafka-operator-0.45.0 ).
2. Attempted to create a Kafka cluster using the provided example YAML file for KRaft mode with dual-role nodes.
3. Applied the YAML configuration using kubectl apply -f examples/kafka/kraft/kafka-with-dual-role-nodes.yaml.
4. Received the following error:
kafkanodepool.kafka.strimzi.io/dual-role created
Error from server (BadRequest): error when creating "examples/kafka/kraft/kafka-with-dual-role-nodes.yaml": Kafka in version "v1beta2" cannot be handled as a Kafka: strict decoding error: unknown field "spec.kafka.metadataVersion"

### Expected behavior

The Kafka CR should be applied successfully without any errors, and the Kafka cluster with KRaft mode and dual-role nodes should be created.

### Strimzi version

0.45.0

### Kubernetes version

1.31

### Installation method

helm chart

### Infrastructure

Bare-metal

### Configuration files and logs

# kubectl apply -f examples/kafka/kraft/kafka-with-dual-role-nodes.yaml
kafkanodepool.kafka.strimzi.io/dual-role created
Error from server (BadRequest): error when creating "examples/kafka/kraft/kafka-with-dual-role-nodes.yaml": Kafka in version "v1beta2" cannot be handled as a Kafka: strict decoding error: unknown field "spec.kafka.metadataVersion"

### Additional context

I have verified that the metadataVersion field is present in the YAML file but not recognized by the current version of the Strimzi Operator.
I am unsure if this field is supported in the v1beta2 API version or if there's another way to configure the metadata version for Kafka clusters using KRaft mode.

**Comment by scholzj** [comment]
I guess that suggest you did not updated the CRDs after upgrading the Strimzi operator? If you use Helm, it will never update the CRDs and you have to do it separately for example using the YAML file with CRDs from our GitHub release. This is a Helm _feature_.
